services/web/server/src/simcore_service_webserver/projects/projects_api.py

```diff
@@ -373,20 +397,22 @@ async def delete_project_node(
     list_running_dynamic_services = await director_v2_api.list_dynamic_services(
         request.app, project_id=f"{project_uuid}", user_id=user_id
     )
-    if any(s["service_uuid"] == node_uuid for s in list_running_dynamic_services):
-        # no need to save the state of the node when deleting it
-        await director_v2_api.stop_dynamic_service(
+
+    fire_and_forget_task(
+        _remove_service_and_its_data_folders(
             request.app,
-            node_uuid,
-            simcore_user_agent=request.headers.get(
+            user_id=user_id,
+            project_uuid=project_uuid,
+            node_uuid=node_uuid,
+            user_agent=request.headers.get(
                 X_SIMCORE_USER_AGENT, UNDEFINED_DEFAULT_SIMCORE_USER_AGENT_VALUE
             ),
-            save_state=False,
-        )
-
-    # remove the node's data if any
-    await storage_api.delete_data_folders_of_project_node(
-        request.app, f"{project_uuid}", node_uuid, user_id
+            stop_service=any(
+                s["service_uuid"] == node_uuid for s in list_running_dynamic_services
```

Comment:
I see you are trying to be smart here. I would actually always try to stop the service and only raise an error if something other than "service is missing" is raised.

Reply:
well sending many rest calls for nothing sounds a bit wasteful no? as discussed I will not do that